src/common/database.cpp

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) rAthena Dev Teams - Licensed under GNU GPL
+// For more information, see LICENCE in the main folder
+
+#include "database.hpp"
+
+#include "showmsg.hpp"
+
+bool nodeExists( const YAML::Node& node, const std::string& name ){
+	try{
+		if( node[name] ){
+			return true;
+		}else{
+			return false;
+		}
+	}catch( const YAML::InvalidNode& ){
+		return false;
+	}
+}
+
+bool Database::verifyCompatability( const YAML::Node& rootNode ){
+	if( !nodeExists( rootNode, "Header" ) ){
+		ShowError( "No database header was found.\n" );
+		return false;
+	}
+
+	const YAML::Node& headerNode = rootNode["Header"];
+
+	if( !nodeExists( headerNode, "Type" ) ){
+		ShowError( "No database type was found.\n" );
+		return false;
+	}
+
+	const YAML::Node& typeNode = headerNode["Type"];
+	const std::string& tmpType = typeNode.as<std::string>();
+
+	if( tmpType != this->type ){
+		ShowError( "Database type mismatch: %s != %s.\n", this->type.c_str(), tmpType.c_str() );
+		return false;
+	}
+
+	uint16 tmpVersion;
+
+	if( !this->asUInt16( headerNode, "Version", &tmpVersion ) ){
+		ShowError("Invalid header version type for %s database.\n", this->type.c_str());
+		return false;
+	}
+
+	if( tmpVersion != this->version ){
+		if( tmpVersion > this->version ){
+			ShowError( "Your database version %hu is not supported by your server. Maximum version is: %hu\n", tmpVersion, this->version );
+			return false;
+		}else if( tmpVersion >= this->minimumVersion ){
+			ShowWarning( "Your database version %hu is outdated and should be updated. Current version is: %hu\n", tmpVersion, this->minimumVersion );
+		}else{
+			ShowError( "Your database version %hu is not supported anymore by your server. Minimum version is: %hu\n", tmpVersion, this->minimumVersion );
+			return false;
+		}
+	}
+
+	return true;
+}
+
+bool Database::load(const std::string& path) {
+	YAML::Node rootNode;
+
+	try {
+		rootNode = YAML::LoadFile(path);
+	}
+	catch(YAML::Exception &e) {
+		ShowError("Failed to read %s database file from '" CL_WHITE "%s" CL_RESET "'.\n", this->type.c_str(), path.c_str());
+		ShowError("%s (%d:%d)\n", e.msg.c_str(), e.mark.line, e.mark.column);
```

Comment:
`ShowError("%s (%d:%d)\n", e.msg.c_str(), e.mark.line, e.mark.column);` this message is not clearly understandable